File: mobile/android/app/src/main/java/com/benimsehrim/app/ui/components/ModernComponents.kt
package com.benimsehrim.app.ui.components

import androidx.compose.animation.core.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.StarHalf
import androidx.compose.material.icons.outlined.StarOutline
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.benimsehrim.app.ui.theme.*

/**
 * Modern gradient button with animation
 */
@Composable
fun GradientButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    loading: Boolean = false,
    gradient: List<Color> = listOf(GradientStart, GradientEnd)
) {
    Button(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .height(56.dp)
            .shadow(
                elevation = if (enabled) 8.dp else 0.dp,
                shape = RoundedCornerShape(16.dp),
                spotColor = gradient.first().copy(alpha = 0.3f)
            ),
        enabled = enabled && !loading,
        colors = ButtonDefaults.buttonColors(
            containerColor = Color.Transparent,
            disabledContainerColor = Color.Gray.copy(alpha = 0.3f)
        ),
        contentPadding = PaddingValues(0.dp),
        shape = RoundedCornerShape(16.dp)
    ) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    brush = Brush.linearGradient(
                        colors = if (enabled) gradient else listOf(Color.Gray, Color.Gray)
                    )
                ),
            contentAlignment = Alignment.Center
        ) {
            if (loading) {
                CircularProgressIndicator(
                    modifier = Modifier.size(24.dp),
                    color = Color.White,
                    strokeWidth = 2.dp
                )
            } else {
                Text(
                    text = text,
                    style = MaterialTheme.typography.titleMedium,
                    color = Color.White
                )
            }
        }
    }
}

/**
 * Modern card with elevation and shadow
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ModernCard(
    modifier: Modifier = Modifier,
    onClick: (() -> Unit)? = null,
    elevation: Int = 2,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        modifier = modifier
            .fillMaxWidth()
            .shadow(
                elevation = elevation.dp,
                shape = RoundedCornerShape(20.dp),
                spotColor = Color.Black.copy(alpha = 0.1f)
            ),
        onClick = onClick ?: {},
        enabled = onClick != null,
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(
            containerColor = SurfaceLight
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            content = content
        )
    }
}

/**
 * Status badge with color coding
 */
@Composable
fun StatusBadge(
    status: String,
    modifier: Modifier = Modifier
) {
    val (backgroundColor, textColor, displayText) = when (status.uppercase()) {
        "PENDING" -> Triple(StatusPending.copy(alpha = 0.15f), StatusPending, "Bekliyor")
        "ACCEPTED" -> Triple(StatusAccepted.copy(alpha = 0.15f), StatusAccepted, "Onaylandı")
        "PREPARING" -> Triple(StatusPreparing.copy(alpha = 0.15f), StatusPreparing, "Hazırlanıyor")
        "READY" -> Triple(StatusReady.copy(alpha = 0.15f), StatusReady, "Hazır")
        "ON_WAY", "DELIVERING" -> Triple(StatusOnWay.copy(alpha = 0.15f), StatusOnWay, "Yolda")
        "DELIVERED" -> Triple(StatusDelivered.copy(alpha = 0.15f), StatusDelivered, "Teslim Edildi")
        "CANCELLED" -> Triple(StatusCancelled.copy(alpha = 0.15f), StatusCancelled, "İptal")
        else -> Triple(Color.Gray.copy(alpha = 0.15f), Color.Gray, status)
    }

    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(8.dp),
        color = backgroundColor
    ) {
        Text(
            text = displayText,
            modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp),
            style = MaterialTheme.typography.labelSmall,
            color = textColor
        )
    }
}

/**
 * Shimmer loading effect
 */
@Composable
fun ShimmerEffect(
    modifier: Modifier = Modifier
) {
    val transition = rememberInfiniteTransition(label = "shimmer")
    val translateAnim by transition.animateFloat(
        initialValue = 0f,
        targetValue = 1000f,
        animationSpec = infiniteRepeatable(
            animation = tween(
                durationMillis = 1200,
                easing = FastOutSlowInEasing
            ),
            repeatMode = RepeatMode.Restart
        ),
        label = "shimmer"
    )

    Box(
        modifier = modifier
            .background(
                brush = Brush.linearGradient(
                    colors = ShimmerColorShades,
                    start = Offset(translateAnim - 1000f, translateAnim - 1000f),
                    end = Offset(translateAnim, translateAnim)
                )
            )
    )
}

/**
 * Empty state view
 */
@Composable
fun EmptyStateView(
    title: String,
    description: String,
    icon: @Composable () -> Unit,
    actionText: String? = null,
    onActionClick: (() -> Unit)? = null
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Box(
            modifier = Modifier
                .size(120.dp)
                .clip(CircleShape)
                .background(Primary.copy(alpha = 0.1f)),
            contentAlignment = Alignment.Center
        ) {
            icon()
        }

        Spacer(modifier = Modifier.height(24.dp))

        Text(
            text = title,
            style = MaterialTheme.typography.titleLarge,
            color = TextPrimary
        )

        Spacer(modifier = Modifier.height(8.dp))

        Text(
            text = description,
            style = MaterialTheme.typography.bodyMedium,
            color = TextSecondary,
            textAlign = androidx.compose.ui.text.style.TextAlign.Center
        )

        if (actionText != null && onActionClick != null) {
            Spacer(modifier = Modifier.height(24.dp))

            GradientButton(
                text = actionText,
                onClick = onActionClick,
                modifier = Modifier.fillMaxWidth(0.6f)
            )
        }
    }
}

/**
 * Rating stars
 */
@Composable
fun RatingStars(
    rating: Double,
    modifier: Modifier = Modifier,
    starSize: Int = 16
) {
    Row(
        modifier = modifier,
        horizontalArrangement = Arrangement.spacedBy(2.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        repeat(5) { index ->
            Icon(
                imageVector = when {
                    index < rating.toInt() -> Icons.Filled.Star
                    index < rating -> Icons.Filled.StarHalf
                    else -> Icons.Outlined.StarOutline
                },
                contentDescription = null,
                modifier = Modifier.size(starSize.dp),
                tint = Warning
            )
        }

        Spacer(modifier = Modifier.width(4.dp))

        Text(
            text = String.format("%.1f", rating),
            style = MaterialTheme.typography.labelSmall,
            color = TextSecondary
        )
    }
}

File: mobile/ios/BenimSehrim/BenimSehrim/Core/Theme/ModernComponents.swift
import SwiftUI

// MARK: - Modern Button Styles

struct GradientButtonStyle: ButtonStyle {
    let gradient: LinearGradient
    let isEnabled: Bool
    var shadowColor: Color
    
    init(gradient: LinearGradient = .primaryGradient, isEnabled: Bool = true, shadowColor: Color = .primary.opacity(0.3)) {
        self.gradient = gradient
        self.isEnabled = isEnabled
        self.shadowColor = shadowColor
    }
    
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(.system(size: 16, weight: .semibold))
            .foregroundColor(.white)
            .frame(maxWidth: .infinity)
            .frame(height: 56)
            .background(
                Group {
                    if isEnabled {
                        gradient
                    } else {
                        Color.gray.opacity(0.3)
                    }
                }
            )
            .cornerRadius(16)
            .shadow(
                color: isEnabled ? shadowColor : .clear,
                radius: 8,
                x: 0,
                y: 4
            )
            .scaleEffect(configuration.isPressed ? 0.98 : 1.0)
            .animation(.spring(response: 0.3), value: configuration.isPressed)
    }
}

// MARK: - Modern Card

struct ModernCard<Content: View>: View {
    let content: Content
    let elevation: CGFloat
    
    init(elevation: CGFloat = 2, @ViewBuilder content: () -> Content) {
        self.elevation = elevation
        self.content = content()
    }
    
    var body: some View {
        content
            .padding(16)
            .background(Color.surfaceLight)
            .cornerRadius(20)
            .shadow(
                color: Color.black.opacity(0.08),
                radius: elevation * 2,
                x: 0,
                y: elevation
            )
    }
}

// MARK: - Status Badge

struct StatusBadge: View {
    let status: String
    
    private var statusInfo: (color: Color, text: String) {
        switch status.uppercased() {
        case "PENDING":
            return (.statusPending, "Bekliyor")
        case "ACCEPTED":
            return (.statusAccepted, "Onaylandı")
        case "PREPARING":
            return (.statusPreparing, "Hazırlanıyor")
        case "READY":
            return (.statusReady, "Hazır")
        case "ON_WAY", "DELIVERING":
            return (.statusOnWay, "Yolda")
        case "DELIVERED":
            return (.statusDelivered, "Teslim Edildi")
        case "CANCELLED":
            return (.statusCancelled, "İptal")
        default:
            return (.gray, status)
        }
    }
    
    var body: some View {
        Text(statusInfo.text)
            .font(.system(size: 11, weight: .medium))
            .foregroundColor(statusInfo.color)
            .padding(.horizontal, 12)
            .padding(.vertical, 6)
            .background(statusInfo.color.opacity(0.15))
            .cornerRadius(8)
    }
}

// MARK: - Shimmer Effect

struct ShimmerView: View {
    @State private var isAnimating = false
    
    var body: some View {
        GeometryReader { geometry in
            Rectangle()
                .fill(
                    LinearGradient(
                        colors: [
                            Color.borderColor,
                            Color.dividerColor,
                            Color.borderColor
                        ],
                        startPoint: .leading,
                        endPoint: .trailing
                    )
                )
                .mask(
                    Rectangle()
                        .fill(
                            LinearGradient(
                                colors: [.clear, .white, .clear],
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                        .rotationEffect(.degrees(70))
                        .offset(x: isAnimating ? geometry.size.width * 2 : -geometry.size.width * 2)
                )
        }
        .onAppear {
            withAnimation(
                Animation.linear(duration: 1.2)
                    .repeatForever(autoreverses: false)
            ) {
                isAnimating = true
            }
        }
    }
}

// MARK: - Empty State View

struct EmptyStateView: View {
    let title: String
    let description: String
    let icon: Image
    let actionText: String?
    let action: (() -> Void)?
    
    init(
        title: String,
        description: String,
        icon: Image,
        actionText: String? = nil,
        action: (() -> Void)? = nil
    ) {
        self.title = title
        self.description = description
        self.icon = icon
        self.actionText = actionText
        self.action = action
    }
    
    var body: some View {
        VStack(spacing: 24) {
            ZStack {
                Circle()
                    .fill(Color.primaryColor.opacity(0.1))
                    .frame(width: 120, height: 120)
                
                icon
                    .resizable()
                    .scaledToFit()
                    .frame(width: 60, height: 60)
                    .foregroundColor(.primaryColor)
            }
            
            VStack(spacing: 8) {
                Text(title)
                    .font(.system(size: 22, weight: .bold))
                    .foregroundColor(.textPrimary)
                
                Text(description)
                    .font(.system(size: 14))
                    .foregroundColor(.textSecondary)
                    .multilineTextAlignment(.center)
            }
            
            if let actionText = actionText, let action = action {
                Button(action: action) {
                    Text(actionText)
                }
                .buttonStyle(GradientButtonStyle())
                .frame(width: 200)
            }
        }
        .padding(32)
    }
}

// MARK: - Rating Stars

struct RatingStars: View {
    let rating: Double
    let starSize: CGFloat
    
    init(rating: Double, starSize: CGFloat = 16) {
        self.rating = rating
        self.starSize = starSize
    }
    
    var body: some View {
        HStack(spacing: 2) {
            ForEach(0..<5) { index in
                Image(systemName: starIcon(for: index))
                    .resizable()
                    .scaledToFit()
                    .frame(width: starSize, height: starSize)
                    .foregroundColor(.warningColor)
            }
            
            Text(String(format: "%.1f", rating))
                .font(.system(size: 11))
                .foregroundColor(.textSecondary)
                .padding(.leading, 4)
        }
    }
    
    private func starIcon(for index: Int) -> String {
        if Double(index) < rating.rounded(.down) {
            return "star.fill"
        } else if Double(index) < rating {
            return "star.leadinghalf.filled"
        } else {
            return "star"
        }
    }
}

// MARK: - Loading Button

struct LoadingButton: View {
    let title: String
    let isLoading: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            if isLoading {
                ProgressView()
                    .progressViewStyle(CircularProgressViewStyle(tint: .white))
            } else {
                Text(title)
            }
        }
        .buttonStyle(GradientButtonStyle(isEnabled: !isLoading))
        .disabled(isLoading)
    }
}

// MARK: - Gradient Text

struct GradientText: View {
    let text: String
    let gradient: LinearGradient
    let font: Font
    
    init(_ text: String, gradient: LinearGradient = .primaryGradient, font: Font = .title) {
        self.text = text
        self.gradient = gradient
        self.font = font
    }
    
    var body: some View {
        Text(text)
            .font(font)
            .foregroundStyle(gradient)
    }
}
